File: core/src/main/java/com/toocol/ssh/core/term/core/ActionCtrlU.java
package com.toocol.ssh.core.term.core;

import com.toocol.ssh.utilities.event.CharEvent;

/**
 * @author ZhaoZhe (joezane.cn@gmail.com)
 * @date 2022/4/25 18:04
 */
public final class ActionCtrlU extends TermCharAction {
    @Override
    public CharEvent[] watch() {
        return new CharEvent[]{CharEvent.CTRL_U};
    }

    @Override
    public boolean act(Term term, CharEvent charEvent, char inChar) {
        term.lineBuilder.delete(0, term.lineBuilder.length());
        term.executeCursorOldX.set(Term.getPromptLen());
        return false;
    }
}

File: core/src/main/java/com/toocol/termio/core/term/core/ActionCtrlU.kt
package com.toocol.termio.core.term.core

import com.toocol.termio.core.term.core.Term.Companion.promptLen
import com.toocol.termio.utilities.event.core.CharEvent

/**
 * @author ZhaoZhe (joezane.cn@gmail.com)
 * @date 2022/4/25 18:04
 */
class ActionCtrlU : TermCharAction() {
    override fun watch(): Array<CharEvent?> {
        return arrayOf(CharEvent.CTRL_U)
    }

    override fun actOnConsole(term: Term, charEvent: CharEvent, inChar: Char): Boolean {
        term.lineBuilder.delete(0, term.lineBuilder.length)
        term.executeCursorOldX.set(promptLen)
        return false
    }

    override fun actOnDesktop(term: Term, charEvent: CharEvent, inChar: Char): Boolean {
        return false
    }
}